threeML/plugins/Fermi_LAT/download_LLE_data.py

```diff
@@ -0,0 +1,278 @@
+
+    rsp_to_get_latest = np.array(_get_latest_version(rsp_to_get))
+
+    lle_to_get_latest = np.array(_get_latest_version(lle_to_get))
+
+    ft2_to_get_latest = np.array(_get_latest_version(ft2_to_get))
+
+
+
+    files_to_download =[]
+    files_existing = []
+
+
+    for i, rsp in enumerate(rsp_to_get_latest):
+
+        if file_existing_and_readable(os.path.join(destination_directory, rsp)):
+
+            files_existing.append(rsp)
+
+            print('Skipping: %s exists in %s' % (rsp, destination_directory))
+
+        else:
+
+            files_to_download.append(rsp)
+
+    for i, lle in enumerate(lle_to_get_latest):
+
+        if file_existing_and_readable(os.path.join(destination_directory, lle)):
+
+            files_existing.append(lle)
+
+            print('Skipping: %s exists in %s' % (lle, destination_directory))
+
+        else:
+
+            files_to_download.append(lle)
+
+    for i, ft2 in enumerate(ft2_to_get_latest):
+
+        if file_existing_and_readable(os.path.join(destination_directory, ft2)):
+
+            files_existing.append(ft2)
+
+            print('Skipping: %s exists in %s' % (ft2, destination_directory))
+
+        else:
+
+            files_to_download.append(ft2)
+
+    # now download the files
+
+    remote_path = "%s/%s/" % (threeML_config['LAT']['public FTP location'], directory)
+
+    download_info = {}
+
+    file_lookup = {'lle':'lle','pt':'ft2','cspec':'rsp'}
+
+    if len(files_to_download) > 0:
+
+        if verbose:
+            print("\nDownloading LLE, RSP and FT2 files...")
+
+        downloaded_files = download_files_from_directory_ftp(remote_path,
+                                                             sanitize_filename(destination_directory),
+                                                             filenames=files_to_download)
+
+        # rsp_files = downloaded_files[:len(rsp_to_get_latest[rsp_filter])]
+        # lle_files = downloaded_files[len(rsp_to_get_latest[rsp_filter]):len(rsp_to_get_latest[rsp_filter]) + len(
+        #         ft2_to_get_latest[ft2_filter])]
+        # ft2_files = downloaded_files[len(rsp_to_get_latest[rsp_filter]) + len(ft2_to_get_latest[ft2_filter]):]
+
+
+
+
+        for download in downloaded_files:
+
+            file_type = _file_type_match.match(download.split("/")[-1]).group(1)
+
+            assert file_type in _valid_file_type, "Something went wrong %s is not an LLE, RSP, or FT2 file" % download.split("/")[-1] #pragma: no cover
+
+            download_info[file_lookup[file_type]] = download
+
+
+
+
+
+
+
+
+
+    for downloaded in files_existing:
+
+        file_type = _file_type_match.match(downloaded).group(1)
+
+        assert file_type in _valid_file_type, "Something went wrong %s is not an LLE, RSP, or FT2 file" % download # pragma: no cover
+
+        download_info[file_lookup[file_type]]= os.path.join(destination_directory, downloaded)
+
+
+
+
+
+
+
+    return download_info
+
+
+def _get_latest_version(filenames):
```

Comment:
Is this exactly the same as in the GBM one? If yes, after I merge, can you fix this by import this function directly from the GBM downloader, instead of copying/paste here?